e2e/LongHaul/device/IotHub.cs

```diff
@@ -304,5 +331,27 @@ private Task<DirectMethodResponse> DirectMethodCallback(DirectMethodRequest meth
             _logger.Trace(unsupportedMessage, TraceSeverity.Warning);
             return Task.FromResult(new DirectMethodResponse(400) { Payload = unsupportedMessage });
         }
+
+        private async Task DesiredPropertyUpdateCallbackAsync(DesiredProperties properties)
+        {
+            var reported = new ReportedProperties();
+            foreach (KeyValuePair<string, object> prop in properties)
+            {
+                // Assume all values are strings.
+                if (!properties.TryGetValue(prop.Key, out string propertyValue))
+                {
+                    _logger.Trace($"Got request for {prop.Key} with non-string [{prop.Value}]", TraceSeverity.Warning);
+                    continue;
+                }
+
+                _logger.Trace($"Got request for {prop.Key} with [{propertyValue}]", TraceSeverity.Information);
+                reported.Add(prop.Key, propertyValue);
+            }
+
+            if (reported.Any())
+            {
+                await _deviceClient.UpdateReportedPropertiesAsync(reported).ConfigureAwait(false);
```

Comment:
metric for # of twin callbacks handled, and how many desired properties handled?